src/thingset_bin.c

```diff
@@ -848,6 +848,7 @@ int thingset_bin_import_data_progressively(struct thingset_context *ts, uint8_t
                         ts->decoder->payload = ts->msg;
                         ts->decoder->elem_count = state.elem_count;
                         *consumed = successfully_parsed_bytes;
+                        *last_id = id;
```

Comment:
Are you sure this is correct? If we parse the ID successfully, but not the value, don't we want to parse the ID again in the next run?